File: frontend/src/types/invoice.ts
export interface Invoice {
  id: number;
  filename: string;
  file_path: string;
  file_type: string;
  vendor?: string;
  invoice_number?: string;
  invoice_date?: string;
  due_date?: string;
  total_amount?: number;
  tax_amount?: number;
  currency: string;
  confidence_score?: number;
  extraction_status: 'pending' | 'success' | 'failed';
  is_processed: boolean;
  is_paid: boolean;
  created_at: string;
}

File: backend/app/schemas/invoice.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class InvoiceBase(BaseModel):
    filename: str
    file_type: str

class InvoiceCreate(InvoiceBase):
    file_path: str

class InvoiceExtractedData(BaseModel):
    vendor: Optional[str]
    invoice_number: Optional[str]
    invoice_date: Optional[datetime]
    due_date: Optional[datetime]
    total_amount: Optional[float]
    tax_amount: Optional[float]
    currency: str = "EUR"
    extracted_text: Optional[str]
    confidence_score: Optional[float]

class InvoiceResponse(InvoiceBase):
    id: int
    file_path: str
    vendor: Optional[str]
    invoice_number: Optional[str]
    invoice_date: Optional[datetime]
    due_date: Optional[datetime]
    total_amount: Optional[float]
    tax_amount: Optional[float]
    currency: str
    confidence_score: Optional[float]
    extraction_status: str
    is_processed: bool
    is_paid: bool
    created_at: datetime

    class Config:
        from_attributes = True
